packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js

```diff
@@ -52,13 +52,16 @@ export default function useNestedSettingsUpdate(
 
 	const { blockListSettings, parentLock } = useSelect(
 		( select ) => {
-			const rootClientId =
-				select( blockEditorStore ).getBlockRootClientId( clientId );
+			const {
+				getBlockRootClientId,
+				getBlockListSettings,
+				getTemplateLock,
+			} = select( blockEditorStore );
+
+			const rootClientId = getBlockRootClientId( clientId );
 			return {
-				blockListSettings:
-					select( blockEditorStore ).getBlockListSettings( clientId ),
-				parentLock:
-					select( blockEditorStore ).getTemplateLock( rootClientId ),
+				blockListSettings: getBlockListSettings( clientId ),
+				parentLock: getTemplateLock( rootClientId ) ?? false,
```

Comment:
I wonder if this should be in the selector itself? I remember I had that change in one PR that I closed at some point.

Reply:
I see you mentioned this alternative in the description of the PR. Let's just do it. I remember @jorgefilipecosta confirmed that both undefined and false are equivalent here.

Reply:
> Performance tests show some significant improvements on this PR. The firstPaint, firstContentfulPaint, loaded and domContentLoaded metrics all go down.

I'd be doubtful about this though. I think it's probably just instability on the PR but we'll confirm it over time in the graph.

Reply:
OK I pushed the alternative implementation where `getTemplateLock` always returns `string | false`, never `undefined`.

> I'd be doubtful about this though. I think it's probably just instability on the PR but we'll confirm it over time in the graph.

Still, an undeniable fact is that the PR removes hundreds of cascading `updateBlockListSettings` dispatches that are running after the initial render.